lib/shrinkwrap.js

```diff
@@ -33,10 +36,34 @@ function shrinkwrap (args, silent, cb) {
   npm.config.set('production', true)
   readPackageTree(dir, andRecalculateMetadata(iferr(cb, function (tree) {
     var pkginfo = treeToShrinkwrap(tree, !!npm.config.get('dev') || /^dev(elopment)?$/.test(npm.config.get('also')))
-    shrinkwrap_(pkginfo, silent, cb)
+
+    readPackage(iferr(cb, function (data) {
+      chain([
+        [lifecycle, data, 'preshrinkwrap'],
+        [shrinkwrap_, pkginfo, silent],
+        [lifecycle, data, 'postshrinkwrap']
+      ], iferr(cb, function (data) {
+        cb(null, data[0])
```

Comment:
The call to `readPacakge` is unnecessary here. `tree.package` contains the same thing as `data`.